analyzers/src/ITs.JsonParser/IssueParser.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.IO;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using ITs.JsonParser.Json;
+using ITs.JsonParser.Reports;
+
+namespace ITs.JsonParser;
+
+/// <summary>
+/// Reads SARIF reports from "output/" and writes custom reports to "actual/".
+/// </summary>
+public static class IssueParser
+{
+    private static readonly Regex RxSonarRule = new("^S[0-9]+$", RegexOptions.Compiled);
+    private static readonly JsonSerializerOptions SerializerOptions = new() { PropertyNameCaseInsensitive = true, WriteIndented = true };
+
+    public static void Execute(string inputRootPath, string outputRootPath)
+    {
+        Console.WriteLine($"Searching for SARIF reports in {inputRootPath}...");
+        var sarifReports = Read(inputRootPath);
+        ConsoleHelper.WriteLineColor($"Successfully parsed {sarifReports.Length} SARIF reports.", ConsoleColor.Green);
+
+        Console.WriteLine("Mapping SARIF reports to Rule reports..");
+        var customReports = sarifReports.SelectMany(ExplodeToRuleIssues).ToArray();
+        ConsoleHelper.WriteLineColor($"Successfully mapped all SARIF reports to {customReports.Length} Rule reports.", ConsoleColor.Green);
+
+        Console.WriteLine($"Writing results to {outputRootPath}");
+        WriteOutput(outputRootPath, customReports);
+        ConsoleHelper.WriteLineColor("Successfully wrote all Rule reports.", ConsoleColor.Green);
+    }
+
+    private static InputReport[] Read(string rootPath) =>
+        Directory.GetFiles(rootPath, "*.json", SearchOption.AllDirectories)
+            .Where(x => !x.Contains("AnalysisWarnings.MsBuild", StringComparison.InvariantCulture))
```

Comment:
The fact that the code needs to know about this filename doesn't scale. Because we will create more files with other well-known names in the future. And that will bite us.

Can we move all the Sarif reports into some sub-directory of `output`? Like `output/Issues`?